### PR TITLE
Cherrypick #34851 "Only wait for cache syncs once in NodeController"

### PR DESCRIPTION
**Pull request by jessfraz**
ping @ncdc @davidopp @gmarek
cherrypick of #34851

---

This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes/34861)

### CODE REVIEW

**Comment by jessfraz** [comment]
the diff is better without whitespace https://github.com/kubernetes/kubernetes/pull/34861/files?w=1

**Comment by davidopp** [comment]
I'm reviewing this now. I just saw the ?w=1 trick, which makes it a lot easier...

**Comment by davidopp** [comment]
LGTM

File: pkg/controller/node/nodecontroller.go [hunk 1]
```diff
@@ -397,110 +397,107 @@ func (nc *NodeController) Run() {
...
+					if remaining < nodeEvictionPeriod {
+						remaining = nodeEvictionPeriod
+					}
+					return false, remaining
+				})
+			}
+		}, nodeEvictionPeriod, wait.NeverStop)
+
+		go wait.Until(func() {
+			if !nc.nodeController.HasSynced() || !nc.podController.HasSynced() || !nc.daemonSetController.HasSynced() {
```
Comment (wojtek-t) [inline pkg/controller/node/nodecontroller.go]: This one should also be removed.
@davidopp

**Pull request merged**